### PR TITLE
SSPN metadata: add migration rules

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/CERNDocumentServer/cds-rdm/issues/128

### CODE REVIEW

File: cds_migrator_kit/rdm/migration/transform/xml_processing/contributors.py [hunk 1]
```diff
@@ -124,11 +128,25 @@ def get_contributor_role(subfield, role, raise_unexpected=False):
...
+
+    return aff_results
+
+
 def extract_json_contributor_ids(info):
     """Extract author IDs from MARC tags."""
     SOURCES = {
         "AUTHOR|(INSPIRE)": "inspire",
         "AUTHOR|(CDS)": "cds",
+        # "AUTHOR|(SzGeCERN)": "CERN", --> ?
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/contributors.py:149]: we should decide if we keep CERN authors IDs

File: cds_migrator_kit/rdm/migration/transform/xml_processing/contributors.py [hunk 2]
```diff
@@ -124,11 +128,25 @@ def get_contributor_role(subfield, role, raise_unexpected=False):
...
 
 
+def get_contributor_affiliations(info):
+    aff_results = []
+    affiliations = force_list(info.get("u", ""))
+
+    service = current_service_registry.get("affiliations")
+
+    for affiliation_name in affiliations:
+        vocabulary_result = service.search(system_identity, params={"q": affiliation_name}).to_dict()
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/contributors.py:138]: how are you ensuring we match a correct affiliation?

File: cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py [hunk 3]
```diff
@@ -71,12 +71,85 @@ def title(self, key, value):
...
+@model.over("description", "^246__")
+def description(self, key, value):
+    """Translates description."""
+    is_abbreviation = value.get("i") == "Abbreviation"
+    abbreviations.append(value.get("a"))
+
+    if is_abbreviation:
+        return "Abbreviations: " + "; ".join(abbreviations)
+
+    return
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py:86]:
```suggestion
    return IgnoreKey
```

File: cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py [hunk 4]
```diff
@@ -71,12 +71,85 @@ def title(self, key, value):
...
+            "description": description_text,
+            "type": {
+                "id": "other",  # what's with the lang
+            }
+        }
+    elif key == "520__":
+        additional_description = {
+            "description": description_text,
+            "type": {
+                "id": "abstract",  # what's with the lang
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py:107]: > what's with the lang
I am not sure what does this comment mean?

File: cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py [hunk 5]
```diff
@@ -71,12 +71,85 @@ def title(self, key, value):
...
+
+@model.over("imprint", "^269__")
+def imprint(self, key, value):
+    """Translates imprint - WARNING - also publisher and publication_date."""
+
+    imprint = {
+        "place": value.get("a"),
+    }
+
+    if not self.get("publication_date"):
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py:122]: this is tricky, you assume that some other field takes precedence. What is the field? In which circumstances can we get the value already assigned to that key before we analyse 269?

File: cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py [hunk 6]
```diff
@@ -121,23 +197,45 @@ def languages(self, key, value):
...
+            if obj not in _subjects:
+                _subjects.append(obj)
+        if subject_e:
+            obj = {"subject": subject_e}
+            if obj not in _subjects:
+                _subjects.append(obj)
+
+    if key == "65017":
+        subject_value = value.get("a")
+        subject = {
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py:222]: how will this be handled later on? if I remember correctly, subjects are supposed to be a vocabulary, but how we will make rdm aware of these values?

File: cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py (hunk 4 above)
Comment (anikachurilova) [reply cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py:107]: Sorry, this comment was more like "notes for myself" than a real question, that's why it looks confusing 😅  but what I meant was that if we create a record in CDS with additional description, it adds these translations of type and I'm not sure whether we want to keep this in the migrated records as well. 
Screenshot 2024-09-16 at 14 28 51

File: cds_migrator_kit/rdm/migration/transform/xml_processing/quality/contributors.py [hunk 7]
```diff
@@ -124,11 +134,39 @@ def get_contributor_role(subfield, role, raise_unexpected=False):
...
+def get_contributor_affiliations(info):
+    aff_results = []
+    affiliations = force_list(info.get("u", ""))
+    vocab_type = "affiliations"
+    service = current_service_registry.get(vocab_type)
+
+    parsed_affiliations = [StringValue(aff, str).parse().filter_regex(ALPHANUMERIC_ONLY) for
+                           aff in affiliations]
+    vocabulary_type = VocabularyType.query.filter_by(id=vocab_type).one()
+    extra_filter = dsl.Q("term", type__id=vocabulary_type.id)
```
Comment (ntarocco) [inline cds_migrator_kit/rdm/migration/transform/xml_processing/quality/contributors.py:146]: isn't it `vocabulary_type.id` `affiliations`? Is the DB query required?

File: setup.cfg [hunk 8]
```diff
@@ -15,7 +15,7 @@ packages = find:
 python_requires = >=3.9
 zip_safe = False
 install_requires =
-    invenio-app-rdm[opensearch2]>=13.0.0b0.dev4
+    invenio-app-rdm[opensearch2]>=13.0.0b1.dev4
```
Comment (ntarocco) [inline setup.cfg:18]: Is this needed given that we install `cds-rdm` below?

**Pull request merged**